File: src/types/certificate.ts
export type CertificateType = 'surah_completion' | 'juz_completion' | 'attendance' | 'competition';

export interface Certificate {
  id: string;
  type: CertificateType;
  title: string;
  detail: string;
  issuedByTeacherName: string | null;
  issuedAt: string;
}

File: backend/app/schemas/certificate.py
from datetime import datetime
from typing import Literal

from app.schemas.base import CamelModel

CertificateType = Literal["surah_completion", "juz_completion", "attendance", "competition"]


class CertificateOut(CamelModel):
    id: str
    type: CertificateType
    title: str
    detail: str
    issued_by_teacher_name: str | None
    issued_at: datetime


class IssueCertificateRequest(CamelModel):
    type: Literal["attendance", "competition"]
    title: str
    detail: str
